deltacat/tests/catalog/test_catalogs.py

```diff
@@ -152,6 +153,49 @@ def test_init_with_default_catalog_name(self, reset_catalogs):
         assert default_catalog.impl == MockCatalogImpl
         assert default_catalog.inner["kwargs"]["id"] == 2
 
+    def test_init_with_config_yaml(self, tmp_path, reset_catalogs):
```

Comment:
Can we also add tests for the modified behavior of `init`, `put_catalog`, and `pop_catalog`?